### PR TITLE
Fixes subview interactive dismiss lag

### LINKED ISSUE #103: Animation lag when using interactive dismissal
I've been experiencing an animation lag (also mentioned in issue #46) when using the InputBarAccessoryView subview method and tracked it down to the way the drag is handled.  
 
The `func handlePanGestureRecognizer(recognizer: UIPanGestureRecognizer)` (in KeyboardManager: 289) calls the `callbacks[.willChangeFrame]?(keyboardNotification)` closure which itself calls the  `func animateAlongside(_ notification: KeyboardNotification, animations: @escaping ()->Void)` from `func bind(inputAccessoryView: UIView) -> Self`.  
 
The lag occurs because the `cachedNotification` has duration of 0.25, which is then used by `animateAlongside` and results in the visual lagging.  
The fix for this issue is simply removing the `animateAlongside` and directly applying the `keyboardHeight` to the constraint constant and calling `layoutIfNeeded`.  

A better fix, I believe, would be to separate the drag gesture handler to its own function (it currently reuses the `keyboardWillChangeFrame(notification:)` that is also used by the corresponding keyboard notification) and either not call `animateAlongside` or call it with a duration of 0s.

I can submit a PR with this fix, but first would like to get some input/opinions regarding this before since it's weird that not many persons are experiencing this.

EDIT: actually an easier fix is just changing `timeInterval` to _var_ in KeyboardNotification and setting it to 0 on the `handlePanGestureRecognizer`

### PR DESCRIPTION
**Pull request by AndrewSB**
Resolves #103

### CODE REVIEW

**Comment by AndrewSB** [comment]
The first commit in here is from #148.  That needs to be merged before this does

File: Sources/KeyboardManager/KeyboardManager.swift [hunk 1]
```diff
@@ -46,6 +46,9 @@ open class KeyboardManager: NSObject, UIGestureRecognizerDelegate {
     
     /// The `NSLayoutConstraintSet` that holds the `inputAccessoryView` to the bottom if its superview
     private var constraints: NSLayoutConstraintSet?
+
+    /// The height of the tab bar passed into `.bind(inputAccessoryView:usingTabBar)` that's used when deciding the y position of the accessory view
+    private var tabBarHeight: CGFloat?
```
Comment (Kaspik) [inline Sources/KeyboardManager/KeyboardManager.swift:51]: remove

File: Sources/KeyboardManager/KeyboardManager.swift [hunk 2]
```diff
@@ -276,14 +278,22 @@ open class KeyboardManager: NSObject, UIGestureRecognizerDelegate {
...
         
         let location = recognizer.location(in: view)
         let absoluteLocation = view.convert(location, to: window)
         var frame = keyboardNotification.endFrame
         frame.origin.y = max(absoluteLocation.y, window.bounds.height - frame.height)
         frame.size.height = window.bounds.height - frame.origin.y
         keyboardNotification.endFrame = frame
-        callbacks[.willChangeFrame]?(keyboardNotification)
+
+        let yCoordinateDirectlyAboveKeyboard = -frame.height + (self.tabBarHeight ?? 0)
```
Comment (Kaspik) [inline Sources/KeyboardManager/KeyboardManager.swift:292]: get rid of tabBarHeight

**Review by Kaspik — CHANGES_REQUESTED** [review]

**Comment by Kaspik** [comment]
Please update this PR so we can merge the animation lag fix. :)

**Comment by AndrewSB** [comment]
updated and rebased!

**Review by Kaspik — APPROVED** [review]
Ideally add changelog entry please :)

**Pull request merged**